openedx/core/djangoapps/notifications/tasks.py

```diff
@@ -70,3 +70,32 @@ def delete_expired_notifications():
         logger.info(f'{delete_count} Notifications deleted in current batch in {time_elapsed} seconds.')
     time_elapsed = datetime.now() - start_time
     logger.info(f'{total_deleted} Notifications deleted in {time_elapsed} seconds.')
+
+
+@shared_task
+@set_code_owner_attribute
+def send_notifications(user_ids, course_key, app_name, notification_type, context, content_url):
+    """
+    Send notifications to the users.
+    """
+    from .models import Notification
+    user_ids = list(set(user_ids))
+
+    # check if what is preferences of user and make decision to send notification or not
+    preferences = CourseNotificationPreference.objects.filter(
+        user_id__in=user_ids,
+        course_id=course_key,
+    )
+    notifications = []
+    for preference in preferences:
+        if preference and preference.get_web_config(app_name, notification_type):
```

Comment:
I think we need to determine what happens when the `web` config is disabled, but the rest of the channels are enabled. What happens then? Is the Notification not created at all?

Reply:
This Implementation is for the web only as objects are created for the Notification model only.  No notification will be created if web is disabled.